include/helpers.h

```diff
@@ -32,6 +32,56 @@
 	static inline _Noreturn unreachable_(void) {}
 #endif
 
+// Use builtins whenever possible, and shim them otherwise
+#ifdef __GNUC__ // GCC or compatible
+	#define ctz __builtin_ctz
+	#define clz __builtin_clz
+
+#elif defined(_MSC_VER)
+	#include <intrin.h>
+	#pragma intrinsic(_BitScanReverse, _BitScanForward)
+	static inline int ctz(unsigned int x)
+	{
+		unsigned long cnt;
+
+		_BitScanReverse(&cnt, x);
+		return cnt;
+	}
+
+	static inline int clz(unsigned int x)
+	{
+		unsigned long cnt;
+
+		_BitScanForward(&cnt, x);
+		return cnt;
+	}
+
+#else
```

Comment:
You could add some define to make this path accessible, makes it testable. Not sure how to call it however.